test/unit/routes/apply/eligibility/claim/test-child-care-details.js

```diff
@@ -0,0 +1,143 @@
+  const COOKIES_EXPIRED = ['apvs-start-application=']
+  const ROUTE = '/apply/eligibility/claim/child-care'
+
+  let app
+
+  let urlPathValidatorStub
+  let expenseUrlRouterStub
+  let insertExpenseStub
+  let childCareExpenseStub
+  let getIsAdvanceClaimStub
+
+  beforeEach(function () {
+    urlPathValidatorStub = sinon.stub()
+    expenseUrlRouterStub = sinon.stub()
+    insertExpenseStub = sinon.stub()
+    childCareExpenseStub = sinon.stub()
+    getIsAdvanceClaimStub = sinon.stub().resolves()
+
+    const route = proxyquire('../../../../../../app/routes/apply/eligibility/claim/child-care-details', {
+      '../../../../services/validators/url-path-validator': urlPathValidatorStub,
+      '../../../../services/routing/expenses-url-router': expenseUrlRouterStub,
+      '../../../../services/data/insert-expense': insertExpenseStub,
+      '../../../../services/domain/expenses/child-care-expense': childCareExpenseStub,
+      '../../../../services/data/get-is-advance-claim': getIsAdvanceClaimStub
+    })
+    app = routeHelper.buildApp(route)
+  })
+
+  describe(`GET ${ROUTE}`, function () {
+    it('should call the URL Path Validator', function () {
+      return supertest(app)
+        .get(ROUTE)
+        .set('Cookie', COOKIES)
+        .expect(function () {
+          sinon.assert.calledOnce(urlPathValidatorStub)
+        })
+    })
+
+    it('should respond with a 200', function () {
+      return supertest(app)
+        .get(ROUTE)
+        .set('Cookie', COOKIES)
+        .expect(200)
+        .expect(function () {
+          sinon.assert.calledOnce(getIsAdvanceClaimStub)
+        })
+    })
+
+    it('should call parseParams', function () {
+      const parseParams = sinon.stub(expenseUrlRouterStub, 'parseParams')
+      return supertest(app)
+        .get(ROUTE)
+        .set('Cookie', COOKIES)
+        .expect(function () {
+          sinon.assert.calledOnce(parseParams)
+        })
+    })
+  })
+
+  describe(`POST ${ROUTE}`, function () {
+    const REDIRECT_URL = 'some-url'
+    const REFRESHMENT_EXPENSE = {}
+
+    it('should call the URL Path Validator', function () {
+      insertExpenseStub.resolves()
+      return supertest(app)
+        .post(ROUTE)
+        .set('Cookie', COOKIES)
+        .expect(function () {
+          sinon.assert.calledOnce(urlPathValidatorStub)
+        })
+    })
+
+    it('should respond with a 302 if domain object is built and then persisted successfully', function () {
+      childCareExpenseStub.returns(REFRESHMENT_EXPENSE)
+      insertExpenseStub.resolves()
+      return supertest(app)
+        .post(ROUTE)
+        .set('Cookie', COOKIES)
+        .expect(function () {
+          sinon.assert.calledOnce(childCareExpenseStub)
+          sinon.assert.calledOnce(insertExpenseStub)
+        })
+        .expect(302)
+    })
+
+    it('should redirect to date-of-birth error page if cookie is expired', function () {
```

Comment:
I guess if session expires it should actually go to the 'who are you visiting' screen (when built!) that precedes date-of-birth?